File: seller/src/main/java/com/example/seller/Adapter/OrderAdapter.java
package com.example.seller.Adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import com.example.seller.Models.FoodRvModel;
import com.example.seller.Models.OrderModal;
import com.example.seller.R;
import com.example.seller.SessionManager;
import com.firebase.ui.database.FirebaseRecyclerAdapter;
import com.firebase.ui.database.FirebaseRecyclerOptions;
import com.google.firebase.database.FirebaseDatabase;

import org.jetbrains.annotations.NotNull;

public class OrderAdapter extends FirebaseRecyclerAdapter<OrderModal, OrderAdapter.ViewHolder> {

    Context context;

    public OrderAdapter(@NonNull @NotNull FirebaseRecyclerOptions<OrderModal> options) {
        super(options);
    }

    @Override
    protected void onBindViewHolder(@NonNull @NotNull ViewHolder holder, int position, @NonNull @NotNull OrderModal model) {

        holder.buyerName.setText(model.getBuyerName());
        holder.buyerAddress.setText(model.getBuyerAddress());
        holder.buyerState.setText(model.getBuyerState());
        holder.buyerPhoneNo.setText(model.getBuyerPhoneNo());
        holder.totalPrice.setText(model.getBuyerTotalPrice());

        String phoneNo = new SessionManager(context).getUserDetailFromSession().get(SessionManager.KEY_PHONE_NO);
        assert phoneNo != null;
        FirebaseRecyclerOptions<FoodRvModel> options = new FirebaseRecyclerOptions.Builder<FoodRvModel>()
                .setQuery(FirebaseDatabase.getInstance().getReference("Seller").child(phoneNo).child("Delivery order")
                        .child(getRef(position).getKey()).child("Food"), FoodRvModel.class)
                .build();
        holder.recyclerView.setLayoutManager(new LinearLayoutManager(context, LinearLayoutManager.VERTICAL, false));
        FoodRvAdapter adapter = new FoodRvAdapter(options);
        holder.recyclerView.setAdapter(adapter);
        adapter.startListening();
    }

    @NonNull
    @NotNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull @NotNull ViewGroup parent, int viewType) {
        context = parent.getContext();
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.sample_order, parent,false);
        return new ViewHolder(view);
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        TextView buyerName, buyerAddress, buyerState, buyerPhoneNo;
        RecyclerView recyclerView;
        TextView totalPrice;

        public ViewHolder(@NonNull @NotNull View itemView) {
            super(itemView);

            buyerName = itemView.findViewById(R.id.buyerName);
            buyerAddress = itemView.findViewById(R.id.buyerAddress);
            buyerState = itemView.findViewById(R.id.buyerState);
            buyerPhoneNo = itemView.findViewById(R.id.buyerPhoneNo);
            recyclerView = itemView.findViewById(R.id.foodRv);
            totalPrice = itemView.findViewById(R.id.totalPrice);

        }
    }
}

File: app/src/main/java/com/example/loginApp/adapters/OrderAdapter.kt
package com.example.loginApp.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.navigation.Navigation
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.example.loginApp.models.OrderModel
import com.example.loginApp.R
import com.example.loginApp.ui.gallery.GalleryFragmentDirections
import com.firebase.ui.database.FirebaseRecyclerAdapter
import com.firebase.ui.database.FirebaseRecyclerOptions

class OrderAdapter(options: FirebaseRecyclerOptions<OrderModel?>) : FirebaseRecyclerAdapter<OrderModel, OrderAdapter.ViewHolder>(options) {

    var context: Context? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        context = parent.context
        val view = LayoutInflater.from(parent.context).inflate(R.layout.sample_order, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int, model: OrderModel) {
        Glide.with(holder.orderShopImage.context).load(model.shopImage).into(holder.orderShopImage)
        holder.orderShopName.text = model.shopName
        holder.orderShopAddress.text = model.shopAddress
        holder.date.text = model.orderDate
        holder.orderPrice.text = model.totalPrice

        holder.itemView.setOnClickListener {
            val action = GalleryFragmentDirections.actionNavGalleryToOrderDetailFragment(getRef(position).key!!)
            Navigation.findNavController(it).navigate(action)
        }
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var orderShopImage : ImageView = itemView.findViewById(R.id.orderShopImage)
        var orderShopName : TextView = itemView.findViewById(R.id.orderShopName)
        var orderShopAddress : TextView = itemView.findViewById(R.id.orderShopAddress)
        var date : TextView = itemView.findViewById(R.id.orderDate)
        var orderPrice : TextView = itemView.findViewById(R.id.orderPrice)
    }
}